src/utils/regex.test.ts

```diff
@@ -47,3 +47,40 @@ describe('NAME_REGEX', () => {
         expect(NAME_REGEX.test('a~a')).toBeFalsy();
     });
 });
+
+describe('SET_NAME_REGEX', () => {
+    it('should allow alphanumeric lower and uppercase', () => {
+        expect(SET_FILTER_NAME_REGEX.test('ThisIsAlphanumeric123')).toBeTruthy();
+    });
+    it(`should allow / @ ( ) ' - _ , . space`, () => {
+        expect(SET_FILTER_NAME_REGEX.test(`( ) ' - _ , . [ ] : |`)).toBeTruthy();
+    });
+    it(`should not allow ; { } ^ ! " # $ % & * + < = > ? \\ ~`, () => {
```

Comment:
I would write a more general title vs `; { } ^ ! " # $ % & * + < = > ? \\ ~`` because if the regex changes you ll have to update the title and will mostly likely forget about it.